src/linear_algebra/GLA.jl

```diff
@@ -156,6 +159,56 @@ plus_lm(x::AbstractVector) = begin
 end
 zero_lm(n::Int) = x::AbstractVector -> zeros(eltype(x), n)
 
+# LinearOps instance
+#-------------------
+
+@auto_hash_equals struct LinearOpDom
+  N::Int
+end
+
+@instance LinearFunctions(LinearOpDom, LinearOperator) begin
+  @import adjoint, +
+
+  dom(f::LinearOperator) = LinearOpDom(size(f,2))
+  codom(f::LinearOperator) = LinearOpDom(size(f,1))
+
+  compose(f::LinearOperator, g::LinearOperator) = g*f
+  id(V::LinearOpDom) = LOs.opEye(V.N)
+
+  oplus(V::LinearOpDom, W::LinearOpDom) = LinearOpDom(V.N + W.N)
+  oplus(f::LinearOperator, g::LinearOperator) = oplus_lo(f, g)
+  mzero(::Type{LinearOpDom}) = LinearOpDom(0)
+  braid(V::LinearOpDom, W::LinearOpDom) = braid_lo(V.N, W.N)
+  mcopy(V::LinearOpDom) =
+    LOs.opExtension(1:V.N, V.N*2)+LOs.opExtension((V.N+1):(V.N*2), V.N*2)
+  delete(V::LinearOpDom) = LOs.opZeros(0, V.N)
+  plus(V::LinearOpDom) =
+    LOs.opRestriction(1:V.N, V.N*2)+LOs.opRestriction((V.N+1):(V.N*2), V.N*2)
+  zero(V::LinearOpDom) = LOs.opZeros(V.N, 0)
+
+  plus(f::LinearOperator, g::LinearOperator) = f+g
+  scalar(V::LinearOpDom, c::Number) = LOs.opEye(typeof(c),V.N)*c
+  antipode(V::LinearOpDom) = scalar(V,-1)
+end
+
+oplus_lo(f::LinearOperator, g::LinearOperator) = begin
```

Comment:
we don't need to suffix the function names here because julia will use multiple dispatch to pick the right method based on the types of the arguments

Reply:
Right, this definition can be moved into the body of the `@instance` with just the name `oplus`.

Reply:
Okay! I was trying to follow the pattern set up by the LinearMaps instance, where there seem to be helper functions (eg delete_lm, plus_lm) defined after the `@instance` block to keep the `@instance` block itself clean. Is it preferred to just have the entire implementation inside the `@instance` block?

Reply:
IMO, it's good to put the implementation inside the `@instance` block when feasible. The reason I had the helper functions in the other case is that I actually had to use them at multiple spots inside the `@instance`, due to the way that `LinearMaps` handles adjoints.

Reply:
Ah, that makes sense! Thanks for the explanation!